tests/trainer/test_trainer.py

```diff
@@ -4232,9 +4232,21 @@ def test_save_best_checkpoint(self):
                     save_strategy="best",
                     compute_metrics=AlmostAccuracy(),
                 )
-
             self.assertIn("`args.metric_for_best_model` must be provided", str(context.exception))
 
+        # Case 4: Metric name not provided and save_best_strategy is "steps" (i.e., not "best").
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
not critical / minor: tbh, it seems a bit out of place for the `test_save_best_checkpoint` (as well as the previous case). I would probably move it into a separate test. Or should it otherwise call at least train and test actual checkpoint saved?

Reply:
I guess I agree that it logically does seem a bit out of place. I think cases 3 and 4 could be grouped into their own methods since the point isn't so much to test the `save_strategy = "best"` itself but more to test the behavior related to `metric_for_best_model`.

I'm not sure if actually running training would be necessary, though. Case 3 is simply to check whether a ValueError is being properly thrown at Trainer initialization time, and case 4 is also simply to check whether the `__post_init__` method of `TrainingArguments` properly initializes `metric_for_best_model` to `"loss"` when `save_strategy != "best"` and `load_best_model_at_end = True`. To me, neither of these seem to require training/evaluation and Trainer instantiation seems sufficient.

Reply:
Agreed, I would also split it into a separate test (or two test). And, yes, we are testing the init here, that's why it was looking out of place.